app/workers/web/push_notification_worker.rb

```diff
@@ -30,26 +39,27 @@ def perform(subscription_id, notification_id)
         'Crypto-Key' => "dh=#{Webpush.encode64(payload.fetch(:server_public_key)).delete('=')};#{web_push_request.crypto_key_header}",
         'Authorization' => web_push_request.authorization_header
       )
+    end
+  end
 
-      request.perform do |response|
-        # If the server responds with an error in the 4xx range
-        # that isn't about rate-limiting or timeouts, we can
-        # assume that the subscription is invalid or expired
-        # and must be removed
-
-        if (400..499).cover?(response.code) && ![408, 429].include?(response.code)
-          @subscription.destroy!
-        elsif !(200...300).cover?(response.code)
-          raise Mastodon::UnexpectedResponseError, response
+  def perform_request(payload)
+    stoplight_wrapper.run do
+      request_pool.with(web_push_request.audience) do |http_client|
+        build_request(http_client, payload).perform do |response|
```

Comment:
Is there anything we can do here for handling something like a firewall block, where the request is just dropped? This is a common failure mode for nfty